proxy/src/main.rs

```diff
@@ -33,13 +36,57 @@ pub struct ProxyConf {
 
     /// control plane address where we would check auth.
     pub cplane_address: SocketAddr,
+
+    pub ssl_config: Option<Arc<rustls::ServerConfig>>,
 }
 
 pub struct ProxyState {
     pub conf: ProxyConf,
     pub waiters: Mutex<HashMap<String, mpsc::Sender<anyhow::Result<DatabaseInfo>>>>,
 }
 
+fn configure_ssl(arg_matches: &ArgMatches) -> anyhow::Result<Option<Arc<rustls::ServerConfig>>> {
+    match (
+        arg_matches.value_of("ssl-key"),
+        arg_matches.value_of("ssl-cert"),
+    ) {
+        (Some(ssl_key_path), Some(ssl_cert_path)) => {
+            let key = {
+                let key_bytes = std::fs::read(ssl_key_path).context("SSL key file")?;
+                let mut keys =
+                    rustls::internal::pemfile::rsa_private_keys(&mut Cursor::new(&key_bytes))
+                        .or_else(|_| {
+                            rustls::internal::pemfile::pkcs8_private_keys(&mut Cursor::new(
+                                &key_bytes,
+                            ))
+                        })
```

Comment:
There's a few of these really long paths that make it a bit harder to see what's going on. I'd add some `use` statements just inside the top of the function -- something like this might do the trick:
```rs
use rustls::internal::pemfile;
use rustls::{ServerConfig, NoClientAuth, ProtocolVersion::TLSv1_3};
use std::fs;
```